R/Rules-methods.R

```diff
@@ -1001,6 +1001,295 @@ setMethod(
   }
 )
 
+## NextBestProbMTDLTE ----
+
+#' @describeIn nextBest find the next best dose based with the highest
+#'  probability of having a toxicity rate less or equal to the target toxicity
+#'  level.
+#'
+#' @aliases nextBest-NextBestProbMTDLTE
+#'
+#' @export
+#' @example examples/Rules-method-nextBest-NextBestProbMTDLTE.R
+#'
+setMethod(
+  f = "nextBest",
+  signature = signature(
+    nextBest = "NextBestProbMTDLTE",
+    doselimit = "numeric",
+    samples = "Samples",
+    model = "GeneralModel",
+    data = "Data"
+  ),
+  definition = function(nextBest, doselimit, samples, model, data, ...) {
+
+    # Matrix with samples from the dose-tox curve at the dose grid points.
+    prob_samples <- sapply(data@doseGrid, prob, model = model, samples = samples)
+
+    # Determine the maximum dose level with a toxicity probability below or
+    # equal to the target and calculate how often a dose is selected as MTD
+    # across iterations.
+    # The first element of the vector is the relative frequency that no
+    # dose in the grid is below or equal to the target, the
+    # second element that the 1st dose is the MTD, etc.
+    prob_mtd_lte <- table(factor(apply(prob_samples <= nextBest@target, 1, sum),
+                                  levels = 0:data@nGrid)) / size(samples)
+
+    # Handling of first element of the vector according to the selected method.
+    # If method = min: first dose = sum of no dose is below target and minimum
+    #   dose is below target
+    # If method = max: last dose = sum of no dose is below target and maximum
+    #   dose is below target
+    allocation_crit <- as.vector(prob_mtd_lte)
+    names(allocation_crit) <- as.character(c(0, data@doseGrid))
+
+    # In case that placebo is used, placebo and no dose are merged
+    if (data@placebo) {
+      allocation_crit[1] <- sum(allocation_crit[1:2])
+      allocation_crit <- allocation_crit[-2]
+    }
+
```

Comment:
I would consider moving the comments 
    # Handling of first element of the vector according to the selected method.
    # If method = min: first dose = sum of no dose is below target and minimum
    #   dose is below target
    # If method = max: last dose = sum of no dose is below target and maximum
    #   dose is below target

in this section just above the if loop that does this as described in the comments and after the if statement for dealing wiht the placebo case.